crates/worker/src/partition/state_machine/command_interpreter.rs

```diff
@@ -100,6 +106,13 @@ pub trait StateReader {
     ) -> BoxStream<'a, Result<(EntryIndex, JournalEntry), restate_storage_api::StorageError>>;
 }
 
+/// Projected [`InvocationStatus`] for cancellation purposes.
+enum InvocationStatusProjection {
```

Comment:
Maybe put this definition somewhere at the bottom of the file, or IDK. Here it's super confusing.

Reply:
Will do.